src/OpenTelemetry/Trace/Export/Abstractions/IExportComponent.cs

```diff
@@ -19,13 +19,8 @@ namespace OpenTelemetry.Trace.Export
     /// <summary>
     /// Component that exports spans.
     /// </summary>
-    public interface IExportComponent
+    public interface IExportComponent : IExportComponentBase
```

Comment:
`IRunningSpanStore` and `ISampledSpanStore ` are still in abstractions as well as this file, is it expected?

Reply:
it will be great to move them out. @trwegner are there still dependencies?

Reply:
The I*SpanStore interfaces have been moved to the OpenTelemetry project from OpenTelemetry.Abstractions.  Rather than having the interfaces in the SDK side we could have just the implementation if that would be more acceptable.  Other option is to simply remove completely since there is not any exporter that is leveraging the functionality.  Thoughts?

Reply:
Oh, I see. We have Abstraction folder in the OpenTelemetry package now. This is misleading. 

I still think having 2 interfaces for everything and base class is not right.
E.g. 
IExportComponentBase
IExportComponent 
ExportComponentBase 
ExportComponent

From my perspective, it's enough that abstract noop ExportComponentBase would declare span stores and we do not need IExportComponent.

Same concern with other interfaces created in this PR.

Reply:
In the latest push I've tried to simplify the interfaces and implementation.  The new folder has been renamed to SpanStore and all related files moved there.  Interface names are reset to the original and the 2nd layer of interface is removed with the additional functionality of those moved into implementation.  The abstract ExportComponentBase is removed since it was really only providing a single method (others were being overriden) which is now moved to ExportComponent.  Downside is that there is now some explicit casting required to access methods that are not defined in IExportComponent.